scisim/api.py

```diff
@@ -0,0 +1,288 @@
+    if error:
+        return error_message(error)
+    username = request.form['username']
+    user = User.query.filter(User.name == username).first()
+    if not user:
+        return error_message("User with username " + username + " was not found")
+
+    user_id = user.id
+    userLoggingIn = Logged_In(user_id = user_id, timestamp=datetime.datetime.now())
+    db.session.add(userLoggingIn)
+    db.session.commit()
+
+    return success_message("User successfully logged in.")
+
+@app.route('/api/users/update_session', methods=["POST"])
+def api_users_update_session():
+    error = check_for_params(["username", "page_id"], request)
+    if error:
+        return error_message(error)
+
+    username = request.form['username']
+    page_id = request.form['page_id']
+
+    user = User.query.filter(User.name == username).first()
+    if not user:
+        return error_message("User with username " + username + " was not found")
+
+    page = Page.query.filter(Page.id == page_id).first()
+    if not page:
+        return error_message("Page with id " + page_id + " was not found")
+
+    user.last_page = page.id
+
+    db.session.commit()
+
+    return success_message("Last page updated successfully")
+
+@app.route('/api/users/last_session', methods=["POST"])
+def api_users_last_session():
+    error = check_for_params(["username"], request)
+    if error:
+        return error_message(error)
+
+    username = request.form["username"]
+
+    user = User.query.filter(User.name == username).first()
+    if not user:
+        return error_message("User with username " + username + " was not found")
+
+    return dumps({"page": user.last_page})
+
+@app.route('/api/users/notes', methods=['POST'])
+def api_users_notes():
+    error = check_for_params(['user_id'], request)
+    if error:
+        return error_message(error)
+
+    return to_json(Note.query.filter(Note.user_id == request.form['user_id']))
+
+@app.route('/api/users/logout', methods=["POST"])
+def api_users_logout():
+    error = check_for_params(["username"], request)
+    if error:
+        return error_message(error)
+    username = request.form['username']
+    user = User.query.filter(User.name == username).first()
+
+    if not user:
+        return error_message("User with username " + username + " was not found")
+
+    loggedInUser = Logged_In.query.filter(Logged_In.user_id == user.id).first()
+    if not loggedInUser:
+        return error_message("User " + user.name + " was not logged in.")
+
+    db.session.delete(loggedInUser)
+    db.session.commit()
+
+    return success_message("User was logged out successfully")
+
+@app.route('/api/groups/create', methods=["POST"])
+def api_groups_create():
+    error = check_for_params(['group_name', 'shared_computer'], request)
+    if error:
+        return error_message(error)
+    group_name = request.form['group_name']
+    shared = request.form['shared_computer']
+    if shared not in ["0", "1", 0, 1]:
```

Comment:
This seems odd, but I guess it's ok. Perhaps define a variable at the top of this file for this weird list.